src/linear_algebra/determinant.lean

```diff
@@ -273,6 +274,13 @@ begin
   simp [update_column_transpose, det_transpose]
 end
 
+/-- `det` is a multilinear map over the rows of the matrix -/
+@[simps apply]
+def det_row_multilinear : multilinear_map R (λ (i : n), n → R) R :=
```

Comment:
Should there also be an equivalent statement that `det` is a multilinear map over the columns of the matrix?

Reply:
I'm not sure that's as easy to express, and is kind of redundant given we have a lemma about the transpose

Reply:
It shouldn't be too hard to express: define `det_column_multilinear := det_row_multilinear.comp_linear_map transpose`, and, if needed, add a `simp` lemma saying `det_column_multilinear M = det M`.

Reply:
`comp_linear_map` won't work though - the reason I can write this lemma is because `matrix n n R` definitionally equal to `(λ (i : n), n → R)` by associating a row to each input. If you want to associate a column to each input, then you have to transpose the input first and use `det_row_multilinear Mᵀ`.

Reply:
```
(@det_row_multilinear n _ _ R _).comp_linear_map transpose
```
gives the error
```
type mismatch at application
  det_row_multilinear.comp_linear_map transpose
term
  transpose
has type
  matrix ?m_1 ?m_2 ?m_5 → matrix ?m_2 ?m_1 ?m_5 : Type (max (max ? ? ?) ? ? ?)
but is expected to have type
  n → ((n → R) →ₗ[R] n → R) : Type (max u v)
```
which is what I would expect

Reply:
Are you looking for this definition?
```
/-- `det` is a multilinear map over the columns of the matrix -/
@[simps apply]
def det_col_multilinear : multilinear_map R (λ (i : n), n → R) R :=
{ to_fun := λ M, det Mᵀ,
  map_add' := λ M i a b, by {
    simp only [det_transpose],
    exact (det_row_multilinear : multilinear_map R _ _).map_add M i a b,
  },
  map_smul' := λ M i c a,by {
    simp only [det_transpose],
    exact (det_row_multilinear : multilinear_map R _ _).map_smul M i c a,
  } }
```
This seems kind of silly to me - a user can just `rw det_transpose` themselves to eliminate the `ᵀ`.

Reply:
Ah, I see. Then I'm happy with this for now.